src/mlpack/methods/ann/rnn_impl.hpp

```diff
@@ -49,6 +49,49 @@ RNN<OutputLayerType, InitializationRuleType, CustomLayers...>::RNN(
   /* Nothing to do here */
 }
 
+template<typename OutputLayerType, typename InitializationRuleType,
+         typename... CustomLayers>
+RNN<OutputLayerType, InitializationRuleType, CustomLayers...>::RNN(
+    const RNN& network) :
+    rho(network.rho),
+    outputLayer(network.outputLayer),
+    initializeRule(network.initializeRule),
+    inputSize(network.inputSize),
+    outputSize(network.outputSize),
+    targetSize(network.targetSize),
+    reset(network.reset),
+    single(network.single),
+    parameter(network.parameter),
+    numFunctions(network.numFunctions),
+    deterministic(network.deterministic)
+{
+  for (size_t i = 0; i < network.network.size(); ++i)
+  {
+    this->network.push_back(boost::apply_visitor(copyVisitor,
+        network.network[i]));
+    boost::apply_visitor(resetVisitor, this->network.back());
+  }
+}
+
+template<typename OutputLayerType, typename InitializationRuleType,
+         typename... CustomLayers>
+RNN<OutputLayerType, InitializationRuleType, CustomLayers...>::RNN(
+    RNN&& network) :
+    rho(std::move(network.rho)),
+    outputLayer(std::move(network.outputLayer)),
+    initializeRule(std::move(network.initializeRule)),
+    inputSize(std::move(network.inputSize)),
+    outputSize(std::move(network.outputSize)),
+    targetSize(std::move(network.targetSize)),
+    reset(std::move(network.reset)),
+    single(std::move(network.single)),
+    parameter(std::move(network.parameter)),
+    numFunctions(std::move(network.numFunctions)),
+    deterministic(std::move(network.deterministic))
+{
+  this->network = std::move(network.network);
```

Comment:
You could put this in the initialization list too if you liked. :+1:

Reply:
Done.